packages/wallet/src/containers/DirectFunding.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { PureComponent } from 'react';
+import { connect } from 'react-redux';
+
+import * as fundingStates from '../redux/states/channels/funding/index';
+import * as actions from '../redux/actions';
+
+import { unreachable } from '../utils/reducer-utils';
+import TransactionFailed from '../components/TransactionFailed';
+import ApproveFunding from '../components/funding/ApproveFunding';
+import { AFundingStep, BFundingStep } from '../components/funding/FundingStep';
+import EtherscanLink from '../components/EtherscanLink';
+
+interface Props {
+  state: fundingStates.FundingState;
+  fundingApproved: () => void;
+  fundingRejected: () => void;
+  fundingSuccessAcknowledged: () => void;
+  fundingDeclinedAcknowledged: () => void;
+  retryTransactionAction: () => void;
+}
+
+class DirectFundingContainer extends PureComponent<Props> {
+  render() {
+    const { state, fundingApproved, fundingRejected, retryTransactionAction } = this.props;
+
+    switch (state.type) {
+      case fundingStates.WAIT_FOR_FUNDING_REQUEST:
+        return null;
+      case fundingStates.WAIT_FOR_FUNDING_APPROVAL:
+        return (
+          <ApproveFunding
+            fundingApproved={fundingApproved}
+            fundingRejected={fundingRejected}
+            requestedTotalFunds={state.requestedTotalFunds}
+            requestedYourContribution={state.requestedYourContribution}
+          />
+        );
+
+      // PlayerA
+      case fundingStates.A_WAIT_FOR_DEPOSIT_TO_BE_SENT_TO_METAMASK:
+        return <AFundingStep step={0} />;
+      case fundingStates.A_SUBMIT_DEPOSIT_IN_METAMASK:
+        return <AFundingStep step={1}>Please confirm the transaction in MetaMask!</AFundingStep>;
+      case fundingStates.A_WAIT_FOR_DEPOSIT_CONFIRMATION:
+        return (
+          <AFundingStep step={2}>
```

Comment:
It looks like this PR did not change this functionality, but leaving this comment anyways... It might be more explicit (and less error prone?) for steps to have names that represent the step functionality as opposed to integer indexes.

Reply:
I'll make an issue for this, so we can deal with it in a separate PR.